src/emailjs-tcp-socket.js

```diff
@@ -619,8 +662,16 @@
                 // process the data available on the socket
                 self._onData(readInfo.data);
 
-                // queue the next read
-                self._readLegacySocket();
+                // Queue the next read.
+                // If a STARTTLS handshake might be upcoming, postpone this onto
+                // the task queue so the IMAP client has a chance to call upgradeToSecure;
+                // without this, we might eat the beginning of the handshake.
+                // If we are already secure, just call it (for performance).
+                if (self.ssl) {     // are we secure yet?
+                    self._readLegacySocket();
+                } else {
+                    setZeroTimeout(self._readLegacySocket.bind(self));
```

Comment:
this is a long shot. i, too, doubt that tcpsockets will ever surface in the web. but google is experimenting with the permissions api, which it now uses e.g. for the web bluetooth api. this is a highly political topic for them (also internally) and they're pretty secretive about that. then there's a couple of advances and ideas around singning mail w/ serviceworker, which would eventually result in something sort-of-similar to what packaged apps deliver, at least in terms of trust. anyway, we'll see what happens ...